hermes/bin/src/event/queue.rs

```diff
@@ -0,0 +1,164 @@
+//! Hermes event queue implementation.
+
+use std::{
+    collections::HashMap,
+    sync::{
+        mpsc::{Receiver, Sender},
+        Arc, Mutex,
+    },
+};
+
+use super::{HermesEvent, HermesEventPayload, TargetApp, TargetModule};
+use crate::{
+    app::HermesAppName,
+    runtime_extensions::state::State,
+    runtime_state::{HermesRuntimeContext, HermesRuntimeState},
+    wasm::module::{Module, ModuleId},
+};
+
+/// Hermes event queue error
+#[derive(thiserror::Error, Debug, Clone)]
+pub(crate) enum Error {
+    /// Target app not found.
+    #[error("Target app not found.")]
+    AppNotFound,
+
+    /// Target module not found.
+    #[error("Target module not found.")]
+    ModuleNotFound,
+
+    /// Failed to add event into the event queue. Event queue is closed.
+    #[error("Failed to add event into the event queue. Event queue is closed.")]
```

Comment:
Just curious, Can other issues prevent events from being added to the event queue?

Reply:
Do you mean like `AnotherEventExecutionLoop` or `ModuleNotFound`/`AppNotFound` ?
No, they are not preventing and not locking for adding new events

Reply:
I mean the problem with the queue itself.

Reply:
See: https://doc.rust-lang.org/std/sync/mpsc/struct.SendError.html
`send` can only fail if the queue is closed.  But even if it fails for another reason, the queue may as well be closed.
The queue is unbounded, so it can't be full.